src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes;
+  color: Colors;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (!this.isValid()) {
+      throw new Error('Not a valid triangle');
+    }
+  }
+
+  isValid(): boolean {
+    const { a, b, c } = this;
+
+    return a > 0
+      && b > 0
+      && c > 0
+      && a < b + c
+      && b < a + c
+      && c < a + b;
```

Comment:
let's separate this logic to two different errors, and make the error text more descriptive